### PR TITLE
[GSoC] Remove XXX replacement on preferences summaries

### PR DESCRIPTION
**Pull request by BrayanDSO**
## Purpose / Description
Use Android's default methods instead + general cleanup + prep for preferences searchBar

## Approach
On the commits (only for reviewing purposes)

## How Has This Been Tested?

https://user-images.githubusercontent.com/69634269/176658171-b360203c-22bf-4dd2-acdf-66565721f36e.mp4

## Checklist
_Please, go through these checks before submitting the PR._

- [X] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [X] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [X] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/res/values/preferences.xml [hunk 1]
```diff
@@ -32,6 +33,10 @@
     <string name="pref_controls_screen_key">controlsScreen</string>
     <string name="pref_reset_languages_key">resetLanguages</string>
     <string name="controls_command_mapping_cat_key">key_map_category</string>
+    <string name="pref_swipe_sensitivity_key">swipeSensitivity</string>
+    <string name="pref_backup_max_key">backupMax</string>
+    <string name="pref_card_browser_font_scale_key">relativeCardBrowserFontSize</string>
```
Comment (david-allison) [inline AnkiDroid/src/main/res/values/preferences.xml:38]: This is not used by the preference

File: AnkiDroid/src/main/res/values/preferences.xml [hunk 2]
```diff
@@ -32,6 +33,10 @@
...
+    <string name="pref_computation_precision_key">advanced_forecast_stats_compute_precision</string>
```
Comment (david-allison) [inline AnkiDroid/src/main/res/values/preferences.xml:39]: This is not used by the preference

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt [hunk 3]
```diff
@@ -992,22 +933,25 @@ class Preferences : AnkiActivity() {
...
+                        CollectionHelper.initializeAnkiDroidDirectory(newPath)
+                        true
+                    } catch (e: StorageAccessException) {
+                        Timber.e(e, "Could not initialize directory: %s", newPath)
+                        MaterialDialog.Builder(requireContext())
+                            .title(R.string.dialog_collection_path_not_dir)
+                            .positiveText(R.string.dialog_ok)
+                            .negativeText(R.string.reset_custom_buttons)
+                            .onPositive { dialog: MaterialDialog, _ -> dialog.dismiss() }
+                            .onNegative { _, _ -> text = CollectionHelper.getDefaultAnkiDroidDirectory(requireContext()) }
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt:951]: `this.text` would be nicer here

File: AnkiDroid/src/main/res/values/10-preferences.xml [hunk 4]
```diff
@@ -17,10 +17,13 @@
 ~
 ~ You should have received a copy of the GNU General Public License along with
 ~ this program.  If not, see <http://www.gnu.org/licenses/>.
---><resources>
+-->
+<resources xmlns:tools="http://schemas.android.com/tools">
```
Comment (david-allison) [inline AnkiDroid/src/main/res/values/10-preferences.xml:21]: revert this

**Review by david-allison — CHANGES_REQUESTED** [review]
I'll be honest, I found this one difficult to grok commit-by-commit. 

Overall, it looks good, and I'd be happy to squash merge it.

File: AnkiDroid/src/main/res/values/preferences.xml [hunk 5]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources xmlns:tools="http://schemas.android.com/tools" tools:ignore="MissingTranslation,DuplicateCrowdInStrings">
+    <string name="pref_summary_percentage">%s\%%</string>
```
Comment (david-allison) [inline AnkiDroid/src/main/res/values/preferences.xml:3]: I don't /think/ this one needs to be translated. Flagging for reviewer 2

Comment (mikehardy) [reply AnkiDroid/src/main/res/values/preferences.xml:3]: LTR. Different symbols. Translate all the things
https://phrase.com/blog/posts/number-localization/#Percentages

Comment (david-allison) [reply AnkiDroid/src/main/res/values/preferences.xml:3]: 😫 Thanks for being right

Comment (mikehardy) [reply AnkiDroid/src/main/res/values/preferences.xml:3]: I wouldn't have looked it up except my natural instict is to be naturally contrary. If you had said "we need to translate this" I would have tried to prove it wrong ;-) (and then been wrong). TIL...

**Review by david-allison — CHANGES_REQUESTED** [review]
```
* What went wrong:
Execution failed for task ':AnkiDroid:processPlayDebugResources'.
> A failure occurred while executing com.android.build.gradle.internal.res.LinkApplicationAndroidResourcesTask$TaskAction
   > Android resource linking failed
     warn: removing resource com.ichi2.anki:string/day_offset_summ without required default value.
     warn: removing resource com.ichi2.anki:string/preference_summary_percentage without required default value.
     warn: removing resource com.ichi2.anki:string/time_limit_summ without required default value.
     com.ichi2.anki.AnkiDroid-main-50:/xml/preferences_advanced.xml:34: error: resource string/preference_summary_literal (aka com.ichi2.anki:string/preference_summary_literal) not found.
     com.ichi2.anki.AnkiDroid-main-50:/xml/preferences_advanced_statistics.xml:22: error: resource string/advanced_forecast_stats_compute_precision (aka com.ichi2.anki:string/advanced_forecast_stats_compute_precision) not found.
     com.ichi2.anki.AnkiDroid-main-50:/xml/preferences_appearance.xml:117: error: resource string/relativeCardBrowserFontSize (aka com.ichi2.anki:string/relativeCardBrowserFontSize) not found.
     error: failed linking file resources.
```

**Comment by BrayanDSO** [comment]
Commits reorganized. Fixed some things as well. PR text updated with new video of me testing the screens

**Comment by BrayanDSO** [comment]
All commits /*should*/ compile now, but testing them one by one is too much time, so I'm fine with squash.

PR video reupdated with last changes. @ankidroid/gsoc-mentor PTAL.

**Comment by mikehardy** [comment]
> All commits /_should_/ compile now, but testing them one by one is too much time, so I'm fine with squash.
> 
> PR video reupdated with last changes. @ankidroid/gsoc-mentor PTAL.

Just to say:

```bash
reset && while [ `git rev-parse HEAD` != "79483d6a3b228a118604fc5488725deaec7ca4a6" ]; do echo not there yet currently at; git rev-parse HEAD; ./gradlew clean uninstallPlayDebug jacocoTestReport :api:lintRelease :AnkiDroid:lintPlayRelease ktlintCheck lint-rules:test; git reset --hard HEAD~1; done
```

nerd power :muscle:  :nerd_face:

**Comment by BrayanDSO** [comment]
> > All commits /_should_/ compile now, but testing them one by one is too much time, so I'm fine with squash.
> > PR video reupdated with last changes. @ankidroid/gsoc-mentor PTAL.
> 
> Just to say:
> 
> ```shell
> reset && while [ `git rev-parse HEAD` != "79483d6a3b228a118604fc5488725deaec7ca4a6" ]; do echo not there yet currently at; git rev-parse HEAD; ./gradlew clean uninstallPlayDebug jacocoTestReport :api:lintRelease :AnkiDroid:lintPlayRelease ktlintCheck lint-rules:test; git reset --hard HEAD~1; done
> ```
> 
> nerd power 💪 🤓

Thanks for the tip, but I still haven't got the time to check them, neither I care about squashing, so I'm fine with it

**Review by david-allison — APPROVED** [review]
LGTM

**Review by lukstbit — APPROVED** [review]

**Pull request merged**